### PR TITLE
fix(modal): make modal a focus trap

### PR DESCRIPTION
**Pull request by dakahn**
ref carbon-components [#1637](https://github.com/ibm/carbon-components/issues/1637), [#1075](https://github.com/ibm/carbon-components/issues/1075), [#1474](https://github.com/IBM/carbon-components/issues/1474)

Adds focus-trap package and implements a focus trap on the modal component. By default the focus trap is active, but users can disable if they _need to_. 

### changed
- implement `<FocusTrap>` component
- set initial focus to first interactable element (close button)

### to test

- observe that by default the modal is a focus trap. Tab moves from the close button to the first button and then to the second and loops back around. shift-tab cycles in reverse.

- in your dev environment set the `focusTrap` prop in Modal-story.js to false to see previous modal focus behavior of being able to move focus out of the modal.

- see if hitting escape while the modal has focus disables focus trapping on the modal

- see if clicking the "outer modal" or the area around the dialogue box does actually blur

attn: @carmacleod, @asudoh

### CODE REVIEW

**Comment by carmacleod** [comment]
@dakahn Awesome! Looking forward to being able to use it in our product.  :)

**Comment by emyarod** [comment]
@dakahn can you update the snapshots to pass CI?

**Comment by dakahn** [comment]
Yeah, sorry, I'm still working on this one -- for some reason it's not letting me assign the WIP label? 😕

File: src/components/Modal/Modal.js [hunk 1]
```diff
@@ -339,18 +348,22 @@ export default class Modal extends Component {
...
-        onClick={this.handleClick}
-        onBlur={this.handleBlur}
-        className={modalClasses}
-        role="presentation"
-        tabIndex={-1}
-        onTransitionEnd={this.props.open ? this.handleTransitionEnd : undefined}
-        ref={this.outerModal}>
-        {modalBody}
-      </div>
+      <FocusTrap active={focusTrap}>
```
Comment (asudoh) [inline src/components/Modal/Modal.js]: Looks that `active` here is for activating focus-wrap behavior. When I close the modal by changing `open` knob at https://deploy-preview-2096--carbon-components-react.netlify.com/experimental/?selectedKind=Modal&selectedStory=Default and manually put e.g. `<input type="text">` into `<body>` by DOM inspector, I can never focus on that `<input>` - That said, did we want:

```suggestion
      <FocusTrap active={open && focusTrap}>
```

?

**Comment by asudoh** [comment]
Looks a cool addition @dakahn!

**Comment by dakahn** [comment]
@asudoh Super sorry I totally acknowledged your review comment and just didn't implement the changes 😿

The fix is in now

**Pull request merged**